api_spec.yaml

```diff
@@ -3246,9 +3246,204 @@ paths:
+                        type: string
+                        enum: ['CONFIG_NOT_FOUND']
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/multitenancy/config/thirdparty/provider/remove:
+    post:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        Delete ThirdParty Tenant Config
+      operationId: deleteTenantConfigPOST
+      parameters:
+        - $ref: '#/components/parameters/multitenancyRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                thirdPartyId:
+                  $ref: '#/components/schemas/thirdPartyId'
+                supertokensTenantId:
+                  $ref: '#/components/schemas/supertokensTenantId'
+
+      responses:
+        200:
+          description: Redirect URL
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: string
+                    example: OK
+                  didConfigExist:
+                    type: boolean
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/multitenancy/config/thirdparty/provider/list:
+    get:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        List all ThirdParty Provider Configs or list ThirdParty Provider Configs filtered by ThirdPartyId or TenantId
+      operationId: listConfigsGET
+      parameters:
+        - $ref: '#/components/parameters/multitenancyRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+        - name: thirdPartyId
+          in: query
+          required: false
+          schema:
+            $ref: '#/components/schemas/thirdPartyId'
+        - name: supertokensTenantId
+          in: query
+          required: false
+          schema:
+            $ref: '#/components/schemas/supertokensTenantId'
+      responses:
+        200:
+          description: ConfigList
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  status:
+                    type: string
+                    example: OK
+                  configs:
+                    type: array
+                    items:
+                      type: object
```

Comment:
all these new APIs should also return a 402 in case tenantId is provided and is != defaultTenantId and they do not have a subscription.

Reply:
Done